### ISSUE TITLE
how to get the map of a certain height

### ISSUE THREAD

**Issue by JWSLAM** [tik0/collision_map_creator_plugin#2]
Hi，
  I wonder whether it is possible to get a map of a certain height？For instance, I want to get map above ground 0.29m which is mouting height of my laser scanner. The map will be used to compare with the SLAM output.Any idea will be highly appreciated! 
  Thanks!

**Comment by tik0** [comment]
Very interesting question and I think it is very related to my first issue #1 which I haven't solved yet. If the objects you want to map start and end within the hight interval of your interest, this [command](https://github.com/tik0/collision_map_creator_plugin#request-to-create-the-collision-map-in-a-height-interval) should do the job.
The problem comes with the ray tracing which I use to create the map. I simply check if I enter an object.
Thus, when the raytracing starts already within the object, no occupied cell is created.
I really want to fix this issue, which is I think just a few lines of code, but I am currently on vacation.

**Comment by JWSLAM** [comment]
Hi tik0，<br>
Thanks for your reply!Wish you a happy vacation!<br>
I have try the command line you proposed. Maybe I do not completely understand your words.<br>
Now I will describe what I want to do.<br>
I have a laser scanner mounting on a robot at the height of 0.29m.<br>
And I put some table in the gazebo world, so at 0.29m the laser scanner will detect the table legs.<br>
If I run SLAM algorithm like gmaping, then in the final map I will see some points which stand for the table legs.<br>
What I want is a perfect map of the height of 0.29m which contains some points claiming the tables legs rather than a rectangle standing for the table surface.<br>
Here comes my questions, is it possible to get such kind of map with your git repo?<br>
Thanks a lot!

**Comment by tik0** [comment]
Yes, totally understand your issue and for now, my implementation is currently not able to do what you want since I check for intersections [here](https://github.com/tik0/collision_map_creator_plugin/blob/master/collision_map_creator.cc#L110). As I have said, it should be just a little fix by a few lines of code exactly at that position. If you'd like to make a proposal, I would be very happy!

**Comment by tik0** [comment]
I've added the fix with the last commit. Now it is possible to map structures at a certain height. Please have a try!
